pipelines/whale/extractor/metric_runner.py

```diff
@@ -92,43 +91,57 @@ def _get_extract_iter(self):
                 for metric_name, metric_details in metric_yaml.items():
                     execution_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
 
-                    try:
-                        query = metric_details["sql"]
-                        query = template_query(query, connection_name=database)
-                        result = list(self.execute(query, is_dict_return_enabled=False))
-
-                        try:
-                            cleaned_result = result[0][0]
-                        except:
-                            cleaned_result = None
-
-                        if "description" in metric_details:
-                            description = metric_details["description"]
-                        else:
-                            description = None
-
-                        if "is_global" in metric_details:
-                            is_global = metric_details["is_global"]
-                        else:
-                            is_global = False
-
-                        yield MetricValue(
-                            database=database,
-                            cluster=cluster,
-                            schema=schema,
-                            table=table,
-                            name=metric_name,
-                            description=description,
-                            execution_time=execution_time,
-                            value=cleaned_result,
-                            is_global=is_global,
-                        )
+                    sql_result = self._compute_sql_result(metric_details, database)
+
+                    if "description" in metric_details:
+                        description = metric_details["description"]
+                    else:
+                        description = None
+
+                    if "is_global" in metric_details:
+                        is_global = metric_details["is_global"]
+                    else:
+                        is_global = False
+
+                    alerts = metric_details["alerts"]
```

Comment:
Can you wrap this in a `if "alerts" in metric_detais` as well? (or something) This should be optional :)